### ISSUE TITLE
Categories for software: visible per default

### ISSUE THREAD

**Issue by cmeessen** [research-software-directory/RSD-as-a-service#992]
Relates to #984 

### Current implementation

The user can add categories by clicking on the drop down menu:

![grafik](https://github.com/research-software-directory/RSD-as-a-service/assets/14222414/0b6f0c35-40f3-4c2b-b185-2850eeb0d754)

Selected categories appear below `<h2>Categories</h2>`

![grafik](https://github.com/research-software-directory/RSD-as-a-service/assets/14222414/fa9c1cb7-68e9-48b0-b7be-565ca64717c1)

### Change request

Administrators can define top level categories that will always appear in the software edit view (*default categories*).

The goal is to nudge users towards using the default categories.

### Example

The defined default category is *Some simple research fruits*:

![grafik](https://github.com/research-software-directory/RSD-as-a-service/assets/14222414/e6b2b3e8-9bf7-4c9b-a155-50ba793b63b3)

**Comment by fembau** [comment]
Some thoughts for further refinement:

1. Only top-level categories could be a default category
2. How default categories and _Categories_ appear and co-exist in the edit-software-UI.
    - cases: no default categories, one default category, several default categories
    - Default categories do not appear in the list/dropdown of _Categories_
    - Currently all headlines have a sub-title. 
3. How default categories and _Categories_ appear and co-exist in the view-software-UI.
    - Appear default categories separated from other categories?
    - ordering of category headlines
4. UI for admin

**Comment by cmeessen** [comment]
Here are some notes from our discussion:

> Only top-level categories could be a default category

Yes

> How default categories and Categories appear and co-exist in the edit-software-UI. 

- cases
  - one or more default categories: default categories sorted alphabetically first, then other categories
- ordering of category headlines/blocks: alphabetically
- sub-titles: add column in db, if empty do not render

> How default categories and Categories appear and co-exist in the view-software-UI. 

- current state of discussion: prefer to move default categories to separate block in the main content because HGF has some really long titles for research programs and topics. We don't know where and how yet.

**Comment by fembau** [comment]
add column for provenance

**Comment by fembau** [comment]
should we add a flag for "where to display" a top-level certain category, e.g. right sidebar (default) or main container (as extra block to emphasize)

**Comment by cmeessen** [comment]
Here is a mockup with our preferred design at the moment. The blocks should be displayed after the Contributors.

![image](https://github.com/research-software-directory/RSD-as-a-service/assets/14222414/1ed9b795-04ef-43b1-8d45-badaaa77620f)

We did not yet decide whether to use the upper or lower representation.

@ctwhome what is your opinion?

**Comment by fembau** [comment]
We further discussed the feature and decided:
1. Categories which are marked as `is_highlight` are visually brought out. In the software edit view they get an extra dropdown menu and in the software view they have an extra section in the main content area.
2. Categories on different levels in the hierarchy could have different extra properties. These are now aggregated in one database column `properties` of type JSONB.
3. In the software edit view: A highlighted category should have a `subtitle` (shown below headline) and a `description` (shown in the blue help/info sidebar) to support the user adding metadata.

**Comment by fembau** [comment]
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/128042473/f342f65e-8b67-4166-ae78-5c6522191a76)

**Comment by ewan-escience** [comment]
Most work is done now in #1227 and #1219 by using highlights. The only thing that is needed is to add the field `tree_level_labels` to the interface, right?

**Comment by cmeessen** [comment]
Yes, as far as I can see the `tree_level_labels` are the only missing feature in the UI.
